frontend/src/components/Budget/TransactionItem.tsx

```diff
@@ -1,32 +1,263 @@
 import * as React from 'react';
-import { Transaction } from 'src/types/budget';
+import {
+  Transaction,
+  TransactionRequest,
+  DeleteTransactions,
+  UpdateTransactions,
+  Category,
+  CategoryOption,
+} from 'src/types/budget';
+import {
+  Formik,
+  FormikProps,
+  FormikValues,
+  Form,
+  Field,
+  FieldProps,
+} from 'formik';
+import { updateTransactions } from 'src/actions/budget';
+import * as Yup from 'yup';
+import { connect } from 'react-redux';
+import { AppState } from 'src/types';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import { ValueType } from 'react-select/lib/types';
+import * as R from 'ramda';
+import Select from 'react-select';
 
 interface Props {
   transaction: Transaction;
+  deleteTransactions: DeleteTransactions;
+  updateTransactions: UpdateTransactions;
+  categories: Category[];
 }
 
-export default class TransactionItem extends React.Component<Props> {
+interface State {
+  isEditing: boolean;
+}
+
+const transactionSchema = Yup.object().shape({
+  categoryId: Yup.number().required(),
+  transactionType: Yup.string()
+    .oneOf(['deposit', 'withdraw'])
+    .required(),
+  amount: Yup.number()
+    .positive('This field must be positive number')
+    .required(),
+  description: Yup.string(),
+  transactionDatetime: Yup.date().required(),
+});
+
+const RadioButton = (fieldProps: FieldProps & FormikValues) => {
+  const {
+    field: { name, value, onChange, onBlur },
+    id,
+    label,
+  } = fieldProps;
+  return (
+    <div>
+      <input
+        name={name}
+        type="radio"
+        value={id}
+        checked={id === value}
+        onChange={onChange}
+        onBlur={onBlur}
+      />
+      <label htmlFor={id}>{label}</label>
+    </div>
+  );
+};
+
+class TransactionItem extends React.Component<Props, State> {
+  state: State = {
+    isEditing: false,
+  };
+
+  handleSubmit = (values: FormikValues) => {
+    const transaction = {} as TransactionRequest;
```

Comment:
You can shorten this code by doing
```suggestion
const { categoryId, description, ... } = values;
const transaction = {
  categoryId,
  description,
  ...
}
    const transaction = {} as TransactionRequest;
```